src/handlers.js

```diff
@@ -119,8 +119,19 @@ const OPERATOR_PRECEDENCE = {
 	'**': 13
 };
 
-/** @type {Record<import('estree').Expression['type'] | 'Super' | 'RestElement', number>} */
+/** @type {Record<import('@typescript-eslint/types').TSESTree.Expression['type'] | 'Super' | 'RestElement', number>} */
 const EXPRESSIONS_PRECEDENCE = {
+	// TODO: add proper precedence
+	ArrayPattern: 1,
+	JSXElement: 1,
+	JSXFragment: 1,
+	ObjectPattern: 1,
+	TSAsExpression: 1,
+	TSInstantiationExpression: 1,
+	TSNonNullExpression: 1,
+	TSSatisfiesExpression: 1,
+	TSTypeAssertion: 1,
+
```

Comment:
I'm not 100% sure what this is supposed to, so im happy if someone could order them for me